raiden/network/transport/udp/udp_utils.py

```diff
@@ -0,0 +1,173 @@
+# -*- coding: utf-8 -*-
+# -*- coding: utf-8 -*-
+import random
+
+import gevent
+from gevent.event import (
+    _AbstractLinkable,
+    Event,
+)
+
+
+def event_first_of(*events):
+    """ Waits until one of `events` is set.
+
+    The event returned is /not/ cleared with any of the `events`, this value
+    must not be reused if the clearing behavior is used.
+    """
+    first_finished = Event()
+
+    if not all(isinstance(e, _AbstractLinkable) for e in events):
+        raise ValueError('all events must be linkable')
+
+    for event in events:
+        event.rawlink(lambda _: first_finished.set())
+
+    return first_finished
+
+
+def timeout_exponential_backoff(retries, timeout, maximum):
+    """ Timeouts generator with an exponential backoff strategy.
+
+    Timeouts start spaced by `timeout`, after `retries` exponentially increase
+    the retry delays until `maximum`, then maximum is returned indefinitely.
+    """
+    yield timeout
+
+    tries = 1
+    while tries < retries:
+        tries += 1
+        yield timeout
+
+    while timeout < maximum:
+        timeout = min(timeout * 2, maximum)
+        yield timeout
+
+    while True:
+        yield maximum
+
+
+def timeout_two_stage(retries, timeout1, timeout2):
```

Comment:
This is used in the RPCClient, so I'm not sure if this is the best place to be. But it's not a bummer.